Automattic-Tracks-iOS/CrashLogging.swift

```diff
@@ -0,0 +1,156 @@
+import Foundation
+import Sentry
+
+/// A class that provides support for logging crashes. Not compatible with Objective-C.
+public class CrashLogging {
+
+    /// A singleton is maintained, but the host application needn't be aware of its existence.
+    internal static let sharedInstance = CrashLogging()
+    fileprivate var dataProvider: CrashLoggingDataProvider?
+    
+    /**
+     Initializes the crash logging system.
+
+     - Parameters:
+     - dataProvider: An object that will provide any required data to the crash logging system.
+
+     - SeeAlso: CrashLoggingDataProvider
+     */
+    public static func start(withDataProvider dataProvider: CrashLoggingDataProvider) {
+
+        // Store the data provider for future use
+        sharedInstance.dataProvider = dataProvider
+
+        // Create a Sentry client and start crash handler
+        do {
+            Client.shared = try Client(dsn: dataProvider.sentryDSN)
+            try Client.shared?.startCrashHandler()
+
+            // Store lots of breadcrumbs to trace errors
+            Client.shared?.breadcrumbs.maxBreadcrumbs = 500
+
+            // Automatically track screen transitions
+            Client.shared?.enableAutomaticBreadcrumbTracking()
+
+            // Automatically track low-memory events
+            Client.shared?.trackMemoryPressureAsEvent()
+
+            // Override event serialization to append the logs, if needed
+            Client.shared?.beforeSerializeEvent = sharedInstance.beforeSerializeEvent
+            Client.shared?.shouldSendEvent = sharedInstance.shouldSendEvent
+
+            // Add additional data
+            Client.shared?.releaseName = dataProvider.releaseName
+            Client.shared?.environment = dataProvider.buildType
+            updateCurrentUser()
+
+        } catch let error {
+            logError(error)
```

Comment:
The only likely exception to come out if this is an invalid `DSN`. If that happens, IMHO I'd rather not crash, and just have this subsystem not work. 

But I could be convinced otherwise :)

Reply:
I just wanted to point out that no logging might actually happen. But it's fine to wait and see how this goes. 😄